tests/linen/module_test.py

```diff
@@ -1024,6 +1024,40 @@ def __call__(self, x):
         return x+1
     self.assertTrue(isinstance(X.Hyper(a=1), X.Hyper))
 
+  def test_sow(self):
+    class Foo(nn.Module):
+      @nn.compact
+      def __call__(self, x):
+        self.sow('intermediates', 'h', x)
+        self.sow('intermediates', 'h', 2 * x)
```

Comment:
Should you add examples with other reducers? Like adding numbers maybe?

Reply:
I don't see how that adds anything number addition is strictly simpler because the order of a + b no longer matters.

Reply:
You're just not currnetly testing the `reduce_fn` and `init_fn` functions -- the unit tests isn't complete.